src/main/java/org/opensearch/ad/ml/ModelManager.java

```diff
@@ -817,20 +819,25 @@ public ThresholdingResult getAnomalyResultForEntity(
                 modelState.setModel(entityModel);
             }
 
-            // trainModelFromExistingSamples may be able to make models not null
-            if (entityModel.getRcf() == null || entityModel.getThreshold() == null) {
-                entityColdStarter.trainModelFromExistingSamples(modelState);
-            }
-
-            if (entityModel.getRcf() != null && entityModel.getThreshold() != null) {
-                return score(datapoint, modelId, modelState);
+            if (entityModel.getErcf().isPresent()) {
+                result = toResult(entityModel.getErcf().get().process(datapoint));
```

Comment:
Design nitpick: the original code (below) uses early returns. The new code here is saying that if an ERCF exists then we will prefer it's output as opposed to the "original model" output. Why not do an early return here? It's allowed since this is the only line that sets `result`. It will decrease the cognitive load, a bit.

Reply:
* yes, ercf will be the current/only model in the system.
* multiple returns are considered less clear than a single return so the new code switches to that. and old code for old models will be removed because with later changes, old models will be converted to ercf before use.

Reply:
Sounds good.

Reply:
1. Can the `process` method support batch data? Asking this as we have historical analysis which run in batch. 
2. Do you have any suggestion about how many data points should be good enough for ERCF to generate results with high confidence.

Reply:
1. No, it only takes one point at a time. For a batch of data points, the client code calls on each of them.
2. Not new numbers yet. The current number should be sufficient.